octopusdeploy3-single-vm-windows/azuredeploy.json

```diff
@@ -0,0 +1,355 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "West US",
+      "allowedValues": [
+        "Central US",
+        "East Asia",
+        "East US",
+        "East US 2",
+        "Japan East",
+        "Japan West",
+        "North Europe",
+        "South Central US",
+        "Southeast Asia",
+        "West Europe",
+        "West US"
+      ],
+      "metadata": {
+        "Description": "Location of the Octopus Deploy server."
+      }
+    },
+    "vmStorageAccountName": {
+      "type": "string",      
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account that will hold the Octopus Deploy Virtual Machine Disks."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "octoadmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "networkDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name used to access the Octopus Deploy server via HTTP or RDP."
+      }
+    },
+    "sqlServerName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the SQL DB Server that will hold the Octopus Deploy data."
+      }
+    },
+    "sqlAdminUsername": {
+      "type": "string",
+      "defaultValue": "sqladmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "sqlAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "licenseFullName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Full Name."
+      }
+    },
+    "licenseOrganisationName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Organisation Name."
+      }
+    },
+    "licenseEmailAddress": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Email Addres."
+      }
+    },
+    "octopusAdminUsername" : {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy web application."
+      }
+    },
+    "octopusAdminPassword" : {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy web application."
+      }
+    }
+  },
+  "variables": {
+    "location": "[parameters('location')]",
```

Comment:
But how is this different from just using "[parameters('location')]" in the template? If you were doing some operations on the parameter value to derive a variable (for e.g. concat, toLower, replace etc.) I can understand the value of using "[variables('location')]" instead of "[parameters('location')]" but I'm not seeing what it accomplishes in this scenario. Looking closely there are other variables following the same pattern as well. 

"licenseFullName": "[parameters('licenseFullName')]",
"licenseOrganisationName": "[parameters('licenseOrganisationName')]",
"licenseEmailAddress": "[parameters('licenseEmailAddress')]",
"octopusAdminUsername": "[parameters('octopusAdminUsername')]",
 "octopusAdminPassword": "[parameters('octopusAdminPassword')]",

Adding @johngossman for his input as well